item type "DicomFragmentSequence" not generated in FellowOakDicom.Serialization.DicomXML class

AshrafGanie opened the pull request:
Fixes # .
XML Serialization issue when dataset element type is DicomFragmentSequence

#### Checklist
- [ ] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch

#### Changes proposed in this pull request:
-Added new fix to handle the DicomFragmentSequence
-Quick new function GetBinaryBase64

MAshraf committed:
Update DicomXML.cs

I am trying to put tother an app to convert DICOM datasets to XML, and I noticed that when a dataset contains elements of type DicomFragmentSequence, such as "Pixel Data," they are not being generated in the output XML.
After looking into the the problem, I took the initiative to address this issue. I have made modifications to the code to handle DicomFragmentSequence elements  include their fragments in the XML output.
I have done initial testing, and the modified code seems to work as expected.

MAshraf committed:
Merge pull request #1 from AshrafGanie/AshrafGanie-patch-1

Update DicomXML.cs

amoerie commented:
Hello @AshrafGanie and thanks for your first contribution!

I have the following remarks:

- While I applaud your enthusiasm, it is usually better to first create an issue so you can describe the problem, and we can have an interactive discussion about the fix. In this specific case though, I agree that the fix is rather straightforward and perhaps did not require a preceding discussion.
- You removed most of our pull request description template. Why? They exist for a reason, for example see my next point
- Could you please include an additional unit test that demonstrates your new code is working?

amoerie commented on FO-DICOM.Core/Serialization/DicomXML.cs:
```diff
@@ -153,6 +171,12 @@ private static string GetBinaryBase64(DicomElement item)
             return Convert.ToBase64String(buffer.Data);
         }
 
+        private static string GetBinaryBase64(byte[] buffer)
```
If this signature was

`private static string GetBinaryBase64(IByteBuffer buffer)`

Then the method could be reused for regular DicomElement instances too.

MAshraf replied:
Hello @amoerie 
 Thanks for your response, let me create the unit test and will also check GetBinaryBase64(IByteBuffer buffer).

AshrafGanie replied:
Hello @amoerie 

Thanks for your response.  I will create required unit test.  Will also look into GetBinaryBase64(IByteBuffer buffer)

MAshraf committed:
Merge branch 'fo-dicom:development' into development

Commit:
Changes made in DiocmXML class to handle DicomFragmentSequence
Signiture of GetBinaryBase64 changed to make it generic for dicom elements
Unit test added to make dure new changes are working expected
